Cargo.toml

```diff
@@ -19,6 +19,7 @@ parking_lot = "0.11.1"
 pin-project = "1.0.1"
 jsonrpsee-proc-macros = { path = "proc-macros" }
 rand = "0.7.3"
+ring-channel = { git = "https://github.com/niklasad1/ring-channel.git", branch = "na-atomicptr" }
```

Comment:
https://github.com/brunocodutra/ring-channel/issues/69 ^^

Reply:
Were you planning to open a PR for this?

Reply:
It's up to the author of the library, he doesn't seem to be very keen on using `AtomicPtr` instead of `NonNullPtr` let's see how it goes. However, the idea is to split this PR into two parts when it works properly

1. Extract `try_send/send.now_or_never` into a separate PR and handle the errors without `ring channel crate`
2. Support for ring channel for ring channel.